src/components/CourseForm/tests/courseForm.test.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { act } from 'react-dom/test-utils';
+import { fireEvent, within } from '@testing-library/react';
+
+import CourseForm from '../CourseForm';
+import renderWithProviders from '../../../utils/renderWithProviders';
+import {
+	ADD_AUTHOR_BTN,
+	AUTHOR_NAME_INPUT,
+	CREATE_AUTHOR_BTN,
+	DELETE_AUTHOR_BTN,
+} from '../../../constants';
+
+const mockedAuthorName = 'Test Author';
+
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useNavigate: () => jest.fn(),
+	useLocation: () => jest.fn(),
+}));
+
+test('should show all authors list', () => {
```

Comment:
Here you can wrap test cases into 'describe' block.
And in it's scope you can place some reusable logic.
Here in each test case you render CourseFrom.
You can create necessary variables in describe block and restore/clear mocks in before/afterEach if needed.
It'll decrease code duplications.

Reply:
The repeatable logic has been moved to the description block